tests/playwright-test/only-changed.spec.ts

```diff
@@ -0,0 +1,344 @@
+  testProcess.clearOutput();
+  testProcess.write('r');
+
+  await testProcess.waitForOutput('b.spec.ts:3:13 › fails');
+  expect(testProcess.output).not.toContain('a.spec');
+});
+
+test('should throw nice error message if git doesnt work', async ({ git, runInlineTest }) => {
+  const result = await runInlineTest({}, { 'only-changed': `this-commit-does-not-exist` });
+
+  expect(result.exitCode).toBe(1);
+  expect(result.output, 'contains our error message').toContain('Cannot detect changed files for --only-changed mode');
+  expect(result.output, 'contains command').toContain('git diff this-commit-does-not-exist --name-only');
+  expect(result.output, 'contains git command output').toContain('unknown revision or path not in the working tree');
+});
+
+test('should suppport component tests', async ({ runInlineTest, git, writeFiles }) => {
+  await writeFiles({
+    'playwright.config.ts': playwrightCtConfigText,
+    'playwright/index.html': `<script type="module" src="./index.ts"></script>`,
+    'playwright/index.ts': `
+    `,
+    'src/contents.ts': `
+      export const content = "Button";
+    `,
+    'src/button.tsx': `
+      import {content} from './contents';
+      export const Button = () => <button>{content}</button>;
+    `,
+    'src/helper.ts': `
+      export { Button } from "./button";
+    `,
+    'src/button.test.tsx': `
+      import { test, expect } from '@playwright/experimental-ct-react';
+      import { Button } from './helper';
+
+      test('pass', async ({ mount }) => {
+        const component = await mount(<Button></Button>);
+        await expect(component).toHaveText('Button');
+      });
+    `,
+    'src/button2.test.tsx': `
+      import { test, expect } from '@playwright/experimental-ct-react';
+      import { Button } from './helper';
+
+      test('pass', async ({ mount }) => {
+        const component = await mount(<Button></Button>);
+        await expect(component).toHaveText('Button');
+      });
+    `,
+  });
+
+  git(`add .`);
+  git(`commit -m "init"`);
+
+  const result = await runInlineTest({}, { 'workers': 1, 'only-changed': true });
+
+  expect(result.exitCode).toBe(1);
+  expect(result.passed).toBe(0);
+  expect(result.output).toContain('No tests found');
+
+  const result2 = await runInlineTest({
+    'src/button2.test.tsx': `
+      import { test, expect } from '@playwright/experimental-ct-react';
+      import { Button } from './helper';
+
+      test('pass', async ({ mount }) => {
+        const component = await mount(<Button></Button>);
+        await expect(component).toHaveText('Different Button');
+      });
+    `
+  }, { 'workers': 1, 'only-changed': true });
+
+  expect(result2.exitCode).toBe(1);
+  expect(result2.failed).toBe(1);
+  expect(result2.output).toContain('button2.test.tsx');
+  expect(result2.output).not.toContain('button.test.tsx');
+
+  git(`commit -am "update button2 test"`);
+
+  const result3 = await runInlineTest({
+    'src/contents.ts': `
```

Comment:
Again, having a test file that does not depend on `src/contents.ts` would be helpful.

Reply:
done in https://github.com/microsoft/playwright/pull/31727/commits/4355c7f8cf45bffd452a46f705287f913f9b93f0